merlin/study/batch.py

```diff
@@ -143,60 +147,70 @@ def batch_worker_launch(spec, com, nodes=None, batch=None):
-    worker_launch = get_yaml_var(batch, "worker_launch", "")
+    launch_pre: Union[str, Dict] = get_yaml_var(batch, "launch_pre", "")
+    launch_args: Union[str, Dict] = get_yaml_var(batch, "launch_args", "")
+    launches: str = get_yaml_var(batch, "worker_launch", "")
 
-    if btype == "flux":
-        launcher = get_batch_type()
-    else:
-        launcher = get_batch_type()
-
-    launchs = worker_launch
-    if not launchs:
-        if btype == "slurm" or launcher == "slurm":
-            launchs = f"srun -N {nodes} -n {nodes}"
-            if bank:
-                launchs += f" -A {bank}"
-            if queue:
-                launchs += f" -p {queue}"
-            if walltime:
-                launchs += f" -t {walltime}"
-        if launcher == "lsf":
-            # The jsrun utility does not have a time argument
-            launchs = f"jsrun -a 1 -c ALL_CPUS -g ALL_GPUS --bind=none -n {nodes}"
-
-    launchs += f" {launch_args}"
+    if not launches:
+        launches = construct_launches(batch, btype, nodes)
+
+    launches += f" {launch_args}"
 
     # Allow for any pre launch manipulation, e.g. module load
     # hwloc/1.11.10-cuda
     if launch_pre:
-        launchs = f"{launch_pre} {launchs}"
-
-    worker_cmd = f"{launchs} {com}"
+        launches = f"{launch_pre} {launches}"
 
+    worker_cmd: str = ""
     if btype == "flux":
-        flux_path = get_yaml_var(batch, "flux_path", "")
-        flux_opts = get_yaml_var(batch, "flux_start_opts", "")
-        flux_exec_workers = get_yaml_var(batch, "flux_exec_workers", True)
+        flux_path: str = get_yaml_var(batch, "flux_path", "")
+        flux_opts: Union[str, Dict] = get_yaml_var(batch, "flux_start_opts", "")
+        flux_exec_workers: Union[str, Dict, bool] = get_yaml_var(batch, "flux_exec_workers", True)
 
-        flux_exec = ""
+        flux_exec: str = ""
         if flux_exec_workers:
             flux_exec = "flux exec"
 
         if "/" in flux_path:
             flux_path += "/"
 
-        flux_exe = os.path.join(flux_path, "flux")
+        flux_exe: str = os.path.join(flux_path, "flux")
 
-        launch = (
-            f"{launchs} {flux_exe} start {flux_opts} {flux_exec} `which {shell}` -c"
+        launch: str = (
+            f"{launches} {flux_exe} start {flux_opts} {flux_exec} `which {shell}` -c"
         )
         worker_cmd = f'{launch} "{com}"'
+    else:
+        worker_cmd = f"{launches} {com}"
 
     return worker_cmd
+
+
+def construct_launches(batch: Dict, btype: Union[str, Dict], nodes: int) -> str:
+    """
+    If no 'worker_launch' is found in the batch yaml, this method constructs the needed data.
+
+    : param batch : (dict): An optional batch override from the worker config
+    : param btype : (str): The type of batch (flux, local, lsf)
+    : param nodes : (int): The number of nodes to use in the batch launch
+    """
+    launches: str = ""
+    launcher: str = get_batch_type()
```

Comment:
'launch string"

Reply:
I don't think so? See [this comment](https://github.com/LLNL/merlin/pull/326#discussion_r701470193), this is logic sorting which distributed workload manager the `launch_string` needs and builds the string differently depending on which is selected, `launcher` holds which DWM is needed.